fix sitemaps

gumaerc opened the pull request:
#### Pre-Flight checklist

- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/673

#### What's this PR do?
First of all, a few prerequisites:
 - Swaps out `hugo-bin` for `hugo-bin-extended`, which uses the latest version of Hugo (0.98.0)
 - Upgrades the version of NodeJS used in Github Actions runners to latest LTS (16.15.0)

This PR accomplishes a few things regarding sitemap generation:
 - General sitemap template moved to the base theme so they apply to all themes
 - Base theme sitemap template now filters out content with `headless` set to true (static JSON, etc.)
 - Adds a new `home.xml` template to the `www` theme that in conjunction with the changes in https://github.com/mitodl/ocw-hugo-projects/pull/176 will facilitate the generation of a `sitemap_index.xml` file.  This file will tell Google where all of the course site sitemaps are located.  It obtains a list of published course sites using the `ocw-studio` API and generates a list of sitemaps based on the response.

#### How should this be manually tested?
 - Clone [`ocw-hugo-projects`](https://github.com/mitodl/ocw-hugo-projects) on the `cg/sitemap-index` branch
 - Clone the [`ocw-www`](https://github.mit.edu/mitocwcontent/ocw-www) content repo
 - Clone `ocw-hugo-themes` on this PR branch and make sure you set the following variables in your `.env` file:
   - `SITEMAP_DOMAIN=ocw.mit.edu`
   - `WWW_HUGO_CONFIG_PATH=/path/to/ocw-hugo-projects/ocw-www/config.yaml`
   - `WWW_CONTENT_PATH=/path/to/mitocwcontent/ocw-www/`
 - Spin up `ocw-www` with `npm run start:www`
 - Visit http://localhost:3000/sitemap.xml and verify that the sitemap index looks correct according to [Google's documentation](https://developers.google.com/search/docs/advanced/sitemaps/large-sitemaps)
 - Visit some of the sitemaps referenced in the index and verify that the links work
 - Visit http://localhost:3000/sitemap-www.xml and verify that the content in `ocw-www` itself is in there, sans `headless` content like instructors
 - Verify that the links inside `sitemap-www.xml` work

FYI: It is expected that the deploy preview will fail as https://github.com/mitodl/ocw-hugo-projects/pull/176 needs to be merged to add the expected env variables to the security configuration

Ferdi commented:
Curious, should we name it sitemap_www.xml  instead of stemap.xml   . Google might get confused which file is witch between sitemap.xml and sitemap_index.xml

Do you have a screenshot to share ?

gumaerc commented:
@Ferdi It's tough to find reliable info on how this works, there's mostly just subjective experiences like this: https://stackoverflow.com/questions/6362942/listing-both-sitemaps-and-sitemap-index-files-in-robots-txt.  From what this post says though, it looks like we might be able to specify `Sitemap: https://ocw.mit.edu/sitemap_index.xml` in `robots.txt` and then make sure `https://ocw.mit.edu/sitemap.xml` is included in the index?  I'll get some locally rendered examples up for you to look at.

ChristopherChudzicki reviewed (approved):
👍 Looks good to me! When using up-to-date ocw-www repo, `sitemap.xml` did not include any headess pages as far as I could tell.

I did not know much about sitemaps before this PR and discussion around it, but having read the google documentation, the index looks like I would expect based on that.

gumaerc commented:
@Ferdi I changed the `robots.txt` to look like this:

```
Sitemap: https://ocw.mit.edu/sitemap_index.xml

User-agent: *

Disallow: /
```

Here's some Gists with my locally generated output:

 - `sitemap_index.xml`: https://gist.github.com/gumaerc/52200fe07ad6cc8b1b0797fd44f700fd
 - `sitemap.xml`: https://gist.github.com/gumaerc/6f1376e45862c7a533c8dd945a176ef1

ChristopherChudzicki commented on package.json:
```diff
@@ -88,7 +88,7 @@
     "gulp": "^4.0.2",
     "hammerjs": "2.0.7",
-    "hugo-bin": "^0.68.0",
+    "hugo-bin-extended": "^0.98.0",
```
@gumaerc Is this updating the hugo version we use in prod? Or just locally in this repo?

gumaerc commented:
@Ferdi You're referring to [this](https://developers.google.com/search/docs/advanced/sitemaps/large-sitemaps) doc?  All of Google's documentation on this is frustratingly non-specific, but that does specify `sitemap_index.xml` as the file name.  In theory, what you suggest should work because `sitemapindex` is part of the XML schema referenced at http://www.sitemaps.org/schemas/sitemap/0.9 and it shouldn't really matter what the filename is.

gumaerc commented:
@ChristopherChudzicki would you mind taking another quick look at this?  I updated this and the `ocw-hugo-projects` PR to generate the files the way Ferdi suggested and updated the testing instructions to reflect what the new filenames should be.
